### PR TITLE
delete requests handle multiple indexes with the "write" alias

### PR DESCRIPTION
**Pull request by NickPapacostas**
I hadn't noticed this issue (nor is it a problem on prod) because deleting seems like it works until you refresh.

### CODE REVIEW

File: thrall/app/lib/ThrallMessageConsumer.scala [hunk 1]
```diff
@@ -52,24 +52,26 @@ object ThrallMessageConsumer extends MessageConsumer(
...
-      } recoverWith {
-        case ImageNotDeletable => {
-          Logger.info(s"Could not delete image $id")
-          Future.successful(EsResponse(s"Image cannot be deleted: $id"))
+  def deleteImage(image: JsValue): Future[List[EsResponse]] =
+    Future.sequence(
+      withImageId(image) { id =>
+        // if we cannot delete the image as it's "protected", succeed and delete
+        // the message anyway.
+        ElasticSearch.deleteImage(id).map { requests =>
```
Comment (NickPapacostas) [inline thrall/app/lib/ThrallMessageConsumer.scala]: So here now we just add an extra mapping layer, and maybe i'm missing something, but i would have thought a flat map here was what I was looking for, but it didn't seem to work.

Comment (NickPapacostas) [inline thrall/app/lib/ThrallMessageConsumer.scala]: or should i be using a for comprehension?

**Comment by NickPapacostas** [comment]
Tested successfully

**Comment by akash1810** [comment]
Its a shame that elastic doesn't handle this for you (deleting from multiple indexes on a single alias).

:+1:

File: thrall/app/lib/ThrallMessageConsumer.scala (hunk 1 above)
Comment (kenoir) [inline thrall/app/lib/ThrallMessageConsumer.scala]: I've found `for` comprehensions can cloud what's happening underneath. I'm not sure why you can't `flatMap` here, what was the problem you were finding?

**Pull request merged**